x/sockopt/sockopt.go

```diff
@@ -50,15 +60,51 @@ var _ HasHopLimit = (*hopLimitOption)(nil)
 
 // TCPOptions represents options for TCP connections.
 type TCPOptions interface {
+	HasWaitUntilBytesAreSent
```

Comment:
We can't have such high-level functionality here. As mentioned below, we should perhaps expose lower level functionality instead.